spec/hyacinth/digital_object/type_def/dynamic_field_data_spec.rb

```diff
@@ -0,0 +1,158 @@
+            'prefLabel' => 'something',
+            'uri' => 'temp:4712538a19c162783874c45c4682fddcf247ac031b06a98912d7df0bc43a3a54'
+          },
+          'role' => [
+            {
+              'term' => {
+                'pref_label' => 'author',
+                'uri' => 'http://id.loc.gov/vocabulary/relators/aut'
+              }
+            }
+          ]
+        }
+      ],
+      'genre' => [
+        {
+          'term' => {
+            'uri' => 'http://vocab.getty.edu/aat/300048715',
+            'other_data' => 'something'
+          }
+        }
+      ]
+    }
+  end
+
+  # Load field definitions and create terms.
+  before do
+    Hyacinth::DynamicFieldsLoader.load_fields!(field_definitions)
+    name_vocab = Vocabulary.create(string_key: 'name', label: 'Name')
+    name_vocab.add_custom_field(field_key: 'uni', label: 'UNI', data_type: 'string')
+    name_vocab.save
+
+    name_role_vocab = Vocabulary.create(string_key: 'name_role', label: 'Name Role')
+    genre_vocab = Vocabulary.create(string_key: 'genre', label: 'Genre')
+
+    Term.create!(pref_label: 'Person, Random', term_type: 'temporary', vocabulary: name_vocab, custom_fields: { uni: 'abc123' })
+    Term.create!(uri: 'http://id.loc.gov/vocabulary/relators/aut', pref_label: 'Author', alt_labels: ['writer'], authority: 'marcrelator', term_type: 'external', vocabulary: name_role_vocab)
+    Term.create!(uri: 'http://vocab.getty.edu/aat/300048715', pref_label: 'articles', authority: 'aat', term_type: 'external', vocabulary: genre_vocab)
+  end
+
+  describe '#from_serialized_form_impl' do
+    let(:expected_serialization) do
+      {
+        'name' => [
+          {
+            'term' => {
+              'pref_label' => 'Person, Random',
+              'uri' => 'temp:4712538a19c162783874c45c4682fddcf247ac031b06a98912d7df0bc43a3a54',
+              'authority' => '',
+              'term_type' => 'temporary',
+              'alt_labels' => [],
+              'uni' => 'abc123'
+            },
+            'role' => [
+              {
+                'term' => {
+                  'pref_label' => 'Author',
+                  'uri' => 'http://id.loc.gov/vocabulary/relators/aut',
+                  'authority' => 'marcrelator',
+                  'term_type' => 'external',
+                  'alt_labels' => ['writer']
+                }
+              }
+            ]
+          }
+        ],
+        'genre' => [
+          {
+            'term' => {
+              'pref_label' => 'articles',
+              'uri' => 'http://vocab.getty.edu/aat/300048715',
+              'authority' => 'aat',
+              'term_type' => 'external',
+              'alt_labels' => []
+            }
+          }
+        ]
+      }
+    end
+
+    it 'adds term data to hash' do
+      expect(type_def.from_serialized_form_impl(descriptive_metadata)).to include(expected_serialization)
+    end
+  end
+
+  describe '#to_serialized_form_impl' do
+    let(:expected_serialization) do
```

Comment:
It would also be good to also include at least one non-controlled-term field in this test to make sure it's serialized as expected.

Reply:
done